smdebug/core/hook.py

```diff
@@ -652,8 +658,10 @@ def _write_scalars(self):
                         scalar_name, scalar_val, self.step, timestamp=timestamp
                     )
             if write_event:
-                if self.writer is None:
-                    self._initialize_writers()
+                if self._is_not_supported():
```

Comment:
if hook is not supported, don't add it to ScalarCache? in save_scalar()

Reply:
Done.